pkg/indexmanagement/scripts.go

```diff
@@ -2,9 +2,11 @@ package indexmanagement
 
 const rolloverScript = `
 set -euo pipefail
+CONNECT_TIMEOUT=${MAX_TIMEOUT:-30.0}
```

Comment:
A few things here:
- I like the idea of allowing a default from the pod env, but I don't think this will work. IIUC the intent is to be able to edit the cronjob and add a `MAX_TIMEOUT` to the spec.jobTemplate.spec.template.spec.containers[].env. But that will get washed away by the operator because of [this](https://github.com/openshift/elasticsearch-operator/blob/91fbd02749632b5c25bb71c8a7cd3841ccbd1e93/pkg/indexmanagement/reconcile.go#L258-L260). And I don't see anywhere else that var could come from.
- Calling the external env var `MAX_TIMEOUT` could be confused with `curl`'s `--max-time` option. Suggest naming it `CONNECT_TIMEOUT` -- same on the outside as on the inside. You can still default it with the same syntax.
- On that note, have you considered supporting `--max-time` also/instead? If the connect succeeds but the server side is hung, the symptom will still occur.
- Nit: `curl` only started accepting decimal values for these timeout options as of version 7.32.0 [1]. I assume that's available in whatever image is being used here -- but nixing the `.0` is safer, and you don't need it.

[1] https://curl.haxx.se/docs/manpage.html#--connect-timeout

Reply:
> A few things here:
> 
> * I like the idea of allowing a default from the pod env, but I don't think this will work. IIUC the intent is to be able to edit the cronjob and add a `MAX_TIMEOUT` to the spec.jobTemplate.spec.template.spec.containers[].env. But that will get washed away by the operator because of [this](https://github.com/openshift/elasticsearch-operator/blob/91fbd02749632b5c25bb71c8a7cd3841ccbd1e93/pkg/indexmanagement/reconcile.go#L258-L260). And I don't see anywhere else that var could come from.

The envvar is a path to allow configuration from the operator.  I would only expect it to be useful in scenarios where an admin has placed the stack in `Unmanaged`.  At this time we are unlikely to promote this into the API and allow changes outside of an `Unmanaged` scenario

> * Calling the external env var `MAX_TIMEOUT` could be confused with `curl`'s `--max-time` option. Suggest naming it `CONNECT_TIMEOUT` -- same on the outside as on the inside. You can still default it with the same syntax.

Not intentional.  Thank you for catching the issue

> * On that note, have you considered supporting `--max-time` also/instead? If the connect succeeds but the server side is hung, the symptom will still occur.

What scenario would you imagine where we can connect but hang?  I believe this job should be relatively quick but my concern is I don't know what a reasonable max is

> * Nit: `curl` only started accepting decimal values for these timeout options as of version 7.32.0 [1]. I assume that's available in whatever image is being used here -- but nixing the `.0` is safer, and you don't need it.
> 
I don't believe we need more then second resolution so taking your suggestion and making an int